spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural_common.py

```diff
@@ -637,12 +658,19 @@ def __write_post_to_pre_table(self, spec, app_vertex, post_slice,
                     subpop_index = np.argwhere(
                         sorted_info_list['lo_atom'] ==
                         pre_vertex_slice.lo_atom).ravel()[0]
-                    masked_sub_pop_index = subpop_index & (2 ** 9 - 1)
+                    masked_sub_pop_index = \
+                        subpop_index & \
+                        SynapseDynamicsStructuralCommon.BIT_MASK_8_BIT
```

Comment:
As above...

Reply:
Copy paste:
`This is correct now (previously a bug). For clarity, the definition of BIT_MASK_X_BIT could be replaced with 0xFF (for 8bit) and 0xFFFF (for 16bit).`